API/projectClass/layersClass.m

```diff
@@ -66,90 +105,125 @@
+            % >>> layers.setLayerValue(2, 2, 'New thickness', paramNames);
+            % 
+            % The same can be achieved using names, to change the 'Thickness' of 'Layer 1'.
+            % 
+            % >>> layers.setLayerValue('Layer 1', 'Thickness', 'New thickness', paramNames);
+            % 
+            % Note that the number of columns change depending on whether ``absorption`` is true or false so the column indices will change also. 
+            % For example, the code below will update 'Roughness' if ``absorption`` is false, otherwise it will update the Imaginary SLD. So it is 
+            % recommended to use column names to ensure the correct change.
+            % 
+            % >>> layers.setLayerValue(2, 4, 'New Roughness', paramNames); 
+            % 
+            % Parameters
+            % ----------
+            % row : string or char array or whole number
+            %     If ``row`` is an integer, it is the row of the layer to update. If it is text, 
+            %     it is the name of the layer to update.
+            % col : string or char array or whole number
+            %     If ``col`` is an integer, it is the column of layer to update. If it is text, 
+            %     it is the name of the column to update. The column names are the following: 'Name', 'Thickness', 
+            %     'SLD', 'Roughness', 'Hydration', 'Hydrate with'. If ``absorption`` is true, the 'SLD' column is replaced 
+            %     with 'SLD Imaginary', and 'SLD Real'.
+            % inputValue : string or char array or whole number
+            %     The name (or row index) of a parameter to replace the one in specified row and column.
+            % paramNames: cell
+            %     A cell array which contains the names of available parameters.
+
             colNames = obj.varTable.Properties.VariableNames;
-            
-            % Find the row index if we have a layer name
-            if isText(row)
-                row = obj.findRowIndex(row, layerNames, 'Unrecognised layer name');
-            elseif isnumeric(row)
-                if (row < 1) || (row > obj.rowCount)
-                    throw(exceptions.indexOutOfRange(sprintf('The row index %d is not within the range 1 - %d', row, obj.rowCount)));
-                end
-            else
-                throw(exceptions.invalidType('Unrecognised layer type'));
-            end
+            row = obj.getValidRow(row);
             
             % Find the column index if we have a column name
             if isText(col)
                 col = obj.findRowIndex(col, colNames, 'Unrecognised column name');
-            elseif isnumeric(col)
-                if (col < 1) || (col > length(colNames))
-                    throw(exceptions.indexOutOfRange(sprintf('The column index %d is not within the range 1 - %d', col, length(colNames))));
+            elseif isnumeric(col) && all(mod(col, 1) == 0)
+                if (col < 2) || (col > length(colNames)) 
```

Comment:
Why are we not able to change the layer name? We can change the name of everything else.

Reply:
I plan to do a refactor for that later